pkg/manager/impl/workflow_manager.go

```diff
@@ -169,8 +169,11 @@ func (w *WorkflowManager) CreateWorkflow(
 			return nil, errors.NewFlyteAdminErrorf(
 				codes.AlreadyExists, "identical workflow already exists with id %v", request.Id)
 		}
-		return nil, errors.NewFlyteAdminErrorf(codes.InvalidArgument,
-			"workflow with different structure already exists with id %v", request.Id)
+		// A workflow exists with different structure - sending much data to help debug on sdk end
+		return nil, errors.NewFlyteAdminErrorf(
+			codes.InvalidArgument,
+			"workflow with different structure already exists with that id: %v. workflow model: %+v.",
```

Comment:
if I'm understanding the original issue correctly, we want flytekit to be able to fetch the existing workflow with an identical digest and then do the comparison on the client-side in order to surface what in the workflow structure has specifically changed.

Right now the `existingMatchingWorkflow` type is an internal db model, with a serialized closure that's not particularly parseable by the flytekit client

I think what we should do is add a proper error type with structure, similar to [these](https://github.com/flyteorg/flyteadmin/blob/master/pkg/errors/errors.go#L82,L109) to store the existing workflow identifier and return this new error in this block. That way the flytekit client can then do a follow-up api call to Get the matching workflow, which will return the [full workflow closure](https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/admin/workflow.proto#L35) in a structured format that flytekit error handling can sensibly use to compare to the workflow definition it is trying to register

Reply:
@katrogan awesome feedback. thanks much! I took another shot at it. Let me know what you think and looks good, then I'll re-do the python end to in this PR: https://github.com/flyteorg/flytekit/pull/1235